### ISSUE TITLE
A question about CmakeList.txt

### ISSUE THREAD

**Issue by Lu-pin-an** [darglein/ADOP#5]
Hello, I am a beginner of neural network. I want to run your program on my computer, but when I Compiling ADOP,in the command "cmake -DCMAKE_PREFIX_PATH="${CONDA}/lib/python3.8/site-packages/torch/;${CONDA}" ..",there was an error,it was "The source directory XXXXdoes not appear to contain CMakeLists.txt."How can i solve this problem?Should i write the CmakeList on myself ?Is there sth wrong in other parts ?this may easy for most people,but i sincerely hope you can answer me,thank you!

**Comment by blurgyy** [comment]
Hi, the code has not been released yet, you are trying to compile an empty code base.  See https://github.com/darglein/ADOP/issues/2#issuecomment-950330204.

**Comment by Lu-pin-an** [comment]
> Hi, the code has not been released yet, you are trying to compile an empty code base. See [#2 (comment)](https://github.com/darglein/ADOP/issues/2#issuecomment-9503302
Thank you for the answer :)!

**Comment by darglein** [comment]
Try again!

**Comment by Lu-pin-an** [comment]
> Try again!

i meet a new problem,when i try to "./install_pytorch.sh",then it said:

   "Building CXX object third_par...base.dir/src/CuptiEventInterface.cpp.o
    FAILED: third_party/kineto/libkineto/CMakeFiles/kineto_base.dir/src/CuptiEventInterface.cpp.o ",

at last it was:"ninja: build stopped: subcommand failed."

the one of possible causes is my ubuntu version is 21.10,should i try in ubuntu 20.04?

**Comment by darglein** [comment]
What is the actual compile error that you are getting?

Please also post the output of `conda -V` and `conda list` inside the adop environment here.

**Comment by Lu-pin-an** [comment]
**1. conda 4.10.1**
**2. environment:**
    # Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             4.5                       1_gnu  
astunparse                1.6.3                      py_0  
blas                      1.0                         mkl  
brotlipy                  0.7.0           py39h27cfd23_1003  
bzip2                     1.0.8                h7b6447c_0  
c-ares                    1.17.1               h27cfd23_0  
ca-certificates           2021.10.26           h06a4308_2  
certifi                   2021.10.8        py39h06a4308_0  
cffi                      1.14.6           py39h400218f_0  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
cmake                     3.19.6               h973ab73_0  
cryptography              35.0.0           py39hd23ed53_0  
cudatoolkit               11.2.72              h2bc3f7f_0    nvidia
cudatoolkit-dev           11.2.2           py39h3811e60_0    conda-forge
cudnn                     8.2.1.32             h86fa8c9_0    conda-forge
dataclasses               0.8                pyh6d0b6a4_7  
expat                     2.4.1                h2531618_2  
future                    0.18.2           py39h06a4308_1  
idna                      3.2                pyhd3eb1b0_0  
intel-openmp              2021.4.0          h06a4308_3561  
krb5                      1.19.2               hac12032_0  
ld_impl_linux-64          2.35.1               h7274673_9  
libcurl                   7.78.0               h0b77cf5_0  
libedit                   3.1.20210714         h7f8727e_0  
libev                     4.33                 h7f8727e_1  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.3.0               h5101ec6_17  
libgomp                   9.3.0               h5101ec6_17  
libnghttp2                1.41.0               hf8bcb03_2  
libssh2                   1.9.0                h1ba5d50_1  
libstdcxx-ng              9.3.0               hd4cf53a_17  
libuv                     1.40.0               h7b6447c_0  
lz4-c                     1.9.3                h295c915_1  
magma-cuda110             2.5.2                         1    pytorch
mkl                       2021.4.0           h06a4308_640  
mkl-include               2021.4.0           h06a4308_640  
mkl-service               2.4.0            py39h7f8727e_0  
mkl_fft                   1.3.1            py39hd3c417c_0  
mkl_random                1.2.2            py39h51133e4_0  
ncurses                   6.3                  h7f8727e_0  
ninja                     1.10.2               hff7bd54_1  
numpy                     1.21.2           py39h20f2e39_0  
numpy-base                1.21.2           py39h79a1101_0  
openssl                   1.1.1l               h7f8727e_0  
pip                       21.2.4           py39h06a4308_0  
pybind11                  2.6.2            py39hff7bd54_1  
pycparser                 2.20                       py_2  
pyopenssl                 21.0.0             pyhd3eb1b0_1  
pysocks                   1.7.1            py39h06a4308_0  
python                    3.9.7                h12debd9_1  
python_abi                3.9                      2_cp39    conda-forge
pyyaml                    6.0              py39h7f8727e_1  
readline                  8.1                  h27cfd23_0  
requests                  2.26.0             pyhd3eb1b0_0  
rhash                     1.4.1                h3c74f83_1  
setuptools                58.0.4           py39h06a4308_0  
six                       1.16.0             pyhd3eb1b0_0  
sqlite                    3.36.0               hc218d9a_0  
tk                        8.6.11               h1ccaba5_0  
typing_extensions         3.10.0.2           pyh06a4308_0  
tzdata                    2021e                hda174b7_0  
urllib3                   1.26.7             pyhd3eb1b0_0  
wheel                     0.37.0             pyhd3eb1b0_1  
xz                        5.2.5                h7b6447c_0  
yaml                      0.2.5                h7b6447c_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.9                haebb681_0  
**3.the Description of the problem is very long，here are a short paragraph of it：**
   /usr/include/generated_cuda_meta.h:1622:5: error: ‘CUuserObject’ does not name a type; did you mean ‘CUsurfObject’?
 1622 |     CUuserObject *object_out;
      |     ^~~~~~~~~~~~
      |     CUsurfObject
/usr/include/generated_cuda_meta.h:1630:5: error: ‘CUuserObject’ does not name a type; did you mean ‘CUsurfObject’?
 1630 |     CUuserObject object;
      |     ^~~~~~~~~~~~
      |     CUsurfObject
/usr/include/generated_cuda_meta.h:1635:5: error: ‘CUuserObject’ does not name a type; did you mean ‘CUsurfObject’?
 1635 |     CUuserObject object;
      |     ^~~~~~~~~~~~
      |     CUsurfObject
/usr/include/generated_cuda_meta.h:1641:5: error: ‘CUuserObject’ does not name a type; did you mean ‘CUsurfObject’?
 1641 |     CUuserObject object;
      |     ^~~~~~~~~~~~
      |     CUsurfObject
/usr/include/generated_cuda_meta.h:1648:5: error: ‘CUuserObject’ does not name a type; did you mean ‘CUsurfObject’?
 1648 |     CUuserObject object;
      |     ^~~~~~~~~~~~
      |     CUsurfObject
/usr/include/generated_cuda_meta.h:1963:5: error: ‘CUflushGPUDirectRDMAWritesTarget’ does not name a type; did you mean ‘cuFlushGPUDirectRDMAWrites_params_st’?
 1963 |     CUflushGPUDirectRDMAWritesTarget target;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |     cuFlushGPUDirectRDMAWrites_params_st
/usr/include/generated_cuda_meta.h:1964:5: error: ‘CUflushGPUDirectRDMAWritesScope’ does not name a type
 1964 |     CUflushGPUDirectRDMAWritesScope scope;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.

**Comment by darglein** [comment]
From `/usr/include/generated_cuda_meta.h` it seems like it's using the system cuda instead of the conda cuda. Try manually activating the conda environment and executing the commands in `install_pytorch.sh`.

Also try to rebuild pytorch by removing the directory `External/pytorch/build`

**Comment by Lu-pin-an** [comment]
it still work,can i install pytorch by my self without using "install_pytorch.sh"?

**Comment by Lu-pin-an** [comment]
it still not work,can i install pytorch by myself without using "install_pytorch.sh"？

**Comment by Lu-pin-an** [comment]
i change a computer and finally sucess:)
